packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -30,9 +30,15 @@ import 'ticker_provider.dart';
 /// See [MaterialPageRoute] for a route that replaces the
 /// entire screen with a platform-adaptive transition.
 abstract class Route<T> {
-  /// The navigator that the route is in, if any.
-  NavigatorState get navigator => _navigator;
-  NavigatorState _navigator;
+  /// The navigator that the route is in, if any.  Deprecated, use navigatorState
+  /// getter instead.
+  @deprecated
+  NavigatorState<BasicNavigator> get navigator => _navigatorState;
+
+  /// The [NavigatorState] of the [BasicNavigator] that the route is in,
+  /// if any.
+  NavigatorState<BasicNavigator> get navigatorState => _navigatorState;
```

Comment:
Because there are still things that need access to the NavigatorState, unfortunately. It just bothered me that it was called "navigator", but what was being returned was _not_ a Navigator.